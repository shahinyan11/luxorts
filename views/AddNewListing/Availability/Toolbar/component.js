import { Button } from 'antd';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import { Navigate } from 'react-big-calendar';

import useCalendarToolbar from 'hooks/calendar/useCalendarToolbar';

import SvgIcon from 'views/shared/SvgIcon';
import GradientButton from 'views/shared/GradientButton';

const Toolbar = ({
  label,
  onNavigate,
  toggleMonthAvailability,
  isFullMonthBlocked,
  isTimePeriodsCreating,
}) => {
  const { navigate } = useCalendarToolbar({ onNavigate });

  return (
    <div className="d-flex justify-content-space-between align-items-center mb-24">
      <div className="d-flex align-items-center">
        <GradientButton
          onClick={navigate(Navigate.PREVIOUS)}
          className="main-btn--icon"
          variant="secondary"
          addonBefore={<SvgIcon icon="arrow-left" />}
        />
        <p className="text-headline-2 ml-16 mr-16 mb-0">{label}</p>
        <GradientButton
          onClick={navigate(Navigate.NEXT)}
          className="main-btn--icon"
          variant="secondary"
          addonBefore={<SvgIcon icon="arrow-right" />}
        />
      </div>

      <Button
        loading={isTimePeriodsCreating}
        onClick={toggleMonthAvailability}
        type="button"
        className="main-text-btn"
      >
        {isFullMonthBlocked && <FormattedMessage id="shared.unblockThisMonth" />}
        {!isFullMonthBlocked && <FormattedMessage id="shared.blockThisMonth" />}
      </Button>
    </div>
  );
};

Toolbar.propTypes = {
  label: PropTypes.string.isRequired,
  onNavigate: PropTypes.func.isRequired,
  toggleMonthAvailability: PropTypes.func.isRequired,
  isFullMonthBlocked: PropTypes.bool.isRequired,
  isTimePeriodsCreating: PropTypes.bool.isRequired,
};

export default Toolbar;
