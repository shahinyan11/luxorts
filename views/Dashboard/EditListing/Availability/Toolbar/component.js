import { Button } from 'antd';
import { FormattedMessage } from 'react-intl';
import PropTypes from 'prop-types';
import { Navigate } from 'react-big-calendar';

import useCalendarToolbar from 'hooks/calendar/useCalendarToolbar';

import SvgIcon from 'views/shared/SvgIcon';
import GradientButton from 'views/shared/GradientButton';

const Toolbar = ({ label, onNavigate }) => {
  const { navigate } = useCalendarToolbar({ onNavigate });

  return (
    <div className="booking-calendar-toolbar mb-24">
      <div className="booking-calendar-toolbar__controls">
        <GradientButton
          onClick={navigate(Navigate.TODAY)}
          className="main-btn--medium mr-16"
          variant="tertiary"
          text={{ id: 'shared.today' }}
        />
        <GradientButton
          onClick={navigate(Navigate.PREVIOUS)}
          className="main-btn--icon main-btn--medium mr-8"
          variant="tertiary"
          addonBefore={<SvgIcon icon="arrow-left" />}
        />
        <GradientButton
          onClick={navigate(Navigate.NEXT)}
          className="main-btn--icon main-btn--medium mr-16"
          variant="tertiary"
          addonBefore={<SvgIcon icon="arrow-right" />}
        />
        <p className="text-headline-2 mb-0">{label}</p>
      </div>
      <div className="booking-calendar-toolbar__links">
        <Button type="button" className="main-text-btn mr-16">
          <FormattedMessage id="shared.availabilitySettings" />
        </Button>
        <Button type="button" className="main-text-btn">
          <FormattedMessage id="shared.pricingSettings" />
        </Button>
      </div>
    </div>
  );
};

Toolbar.propTypes = {
  label: PropTypes.string.isRequired,
  onNavigate: PropTypes.func.isRequired,
};

export default Toolbar;
