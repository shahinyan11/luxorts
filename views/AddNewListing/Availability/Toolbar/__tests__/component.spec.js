import { shallow } from 'enzyme';

import Toolbar from '../component';

const mockedCalendarToolbarData = {
  navigate: jest.fn(() => jest.fn()),
};
jest.mock('hooks/calendar/useCalendarToolbar', () => jest.fn(() => mockedCalendarToolbarData));

describe('Toolbar component', () => {
  const props = {
    label: 'label',
    onNavigate: jest.fn(),
    toggleMonthAvailability: jest.fn(),
    isFullMonthBlocked: false,
    isTimePeriodsCreating: false,
  };

  const component = shallow(<Toolbar {...props} />);

  it('matches snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('matches snapshot when isFullMonthBlocked equals true', () => {
    component.setProps({
      isFullMonthBlocked: true,
    });

    expect(component).toMatchSnapshot();
  });
});
