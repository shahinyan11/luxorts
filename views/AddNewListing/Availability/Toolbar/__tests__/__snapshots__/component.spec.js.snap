// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toolbar component matches snapshot 1`] = `
<div
  className="d-flex justify-content-space-between align-items-center mb-24"
>
  <div
    className="d-flex align-items-center"
  >
    <GradientButton
      addonAfter={null}
      addonBefore={
        <SvgIcon
          icon="arrow-left"
        />
      }
      className="main-btn--icon"
      onClick={[MockFunction]}
      variant="secondary"
    />
    <p
      className="text-headline-2 ml-16 mr-16 mb-0"
    >
      label
    </p>
    <GradientButton
      addonAfter={null}
      addonBefore={
        <SvgIcon
          icon="arrow-right"
        />
      }
      className="main-btn--icon"
      onClick={[MockFunction]}
      variant="secondary"
    />
  </div>
  <Button
    className="main-text-btn"
    loading={false}
    onClick={[MockFunction]}
    type="button"
  >
    <Memo(FormattedMessage)
      id="shared.blockThisMonth"
    />
  </Button>
</div>
`;

exports[`Toolbar component matches snapshot when isFullMonthBlocked equals true 1`] = `
<div
  className="d-flex justify-content-space-between align-items-center mb-24"
>
  <div
    className="d-flex align-items-center"
  >
    <GradientButton
      addonAfter={null}
      addonBefore={
        <SvgIcon
          icon="arrow-left"
        />
      }
      className="main-btn--icon"
      onClick={[MockFunction]}
      variant="secondary"
    />
    <p
      className="text-headline-2 ml-16 mr-16 mb-0"
    >
      label
    </p>
    <GradientButton
      addonAfter={null}
      addonBefore={
        <SvgIcon
          icon="arrow-right"
        />
      }
      className="main-btn--icon"
      onClick={[MockFunction]}
      variant="secondary"
    />
  </div>
  <Button
    className="main-text-btn"
    loading={false}
    onClick={[MockFunction]}
    type="button"
  >
    <Memo(FormattedMessage)
      id="shared.unblockThisMonth"
    />
  </Button>
</div>
`;
